ground/src/main/java/com/google/android/ground/ui/map/Feature.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.ui.map
+
+import com.google.android.ground.model.geometry.Geometry
+
+/** Represents an individual feature on a map with a given [Geometry]. */
+data class Feature(val tag: Tag, val geometry: Geometry) {
+  /** Denotes the kind of entity this map feature represents. */
+  sealed class Tag(open val id: String)
+  /** A [Feature.Tag] that indicates this map feature represents a location of interest. */
+  data class LocationOfInterestTag(
```

Comment:
Is it possible to abstract away LOIs completely from the map UI layer? Imo the map adapter should deal exclusively with map concepts like features, geometries, and UI events.

Reply:
We could just do `Feature` w/ `id, title, caption`... -- we'd presumably still need *some* kind of tag to indicate what kind of item it is. Note that we don't actually refer to LOIs directly at all outside of using the name in this tag type. (and the VM which converts LOis to Features)

Reply:
Though, I guess it's a YAGNI -- at the moment we really *only* have LOI features so we could just make this a data class with a few fields

Reply:
There are a few geometries besides LOIs that the map also needs to render in MVP:

- Point and polygon geometries submitted as part of data collection (Submissions)
- A LineString with handles on vertices representing the interim state while drawing polygons
- A pin at the current location when dropping a pin

In the future there will probably be more, but YAGNI for those.

Reply:
do we think something like:

```kotlin
data class Feature(val metadata: Metadata, val geometry: Geometry) {
  data class Metadata(
    val identifier: String
    val title: String?
    val modificationInfo: AuditInfo 
  )
}
```

Is generic enough to capture all of those cases? Otherwise we could also make the class generic and take various bundles for different types

Reply:
What does the UI layer need this Metadata for?  Iiuc a unique key (id and type), geometry, and some styling info should be sufficient.

Reply:
Ah you're right, I thought we needed the `title` and `modificationInfo` bits for cluster items but they are optional -- and we're just rendering a custom icon anyway. -- ok lemme refactor this a bit